### PR TITLE
Add trailing commas section to style guide

### PR DESCRIPTION
**Pull request by diegopetrucci**

### CODE REVIEW

**Review by zzcgumn — APPROVED** [review]

**Review by AliSoftware — APPROVED** [review]
Approved since looking at our existing codebase, that's indeed the rule we have applied so far in most places.

**Review by RuiAAPeres — APPROVED** [review]
Can we extend this to other things besides Arrays (e.g. dictionaries)?

**Comment by diegopetrucci** [comment]
@RuiAAPeres the swift lint rule fixes dictionaries too, so might as well https://github.com/realm/SwiftLint/blob/master/Rules.md#trailing-comma

File: Cookbook/Style-guide/README.md [hunk 1]
```diff
@@ -958,6 +959,28 @@ In larger expressions, optional parentheses can sometimes make code read more cl
 let playerMark = (player == current ? "X" : "O")
 ```
 
+## Commas
+
+Trailing commas in arrays and dictionaries should be avoided.
```
Comment (mluisbrown) [inline Cookbook/Style-guide/README.md]: I fundamentally disagree with this.

**Review by mluisbrown — CHANGES_REQUESTED** [review]
I fundamentally disagree with removing trailing commas. They are a very useful feature to allow easy refactoring.
I'm not suggesting we should enforce them everywhere, but I see no reason at all to force their removal.

File: Cookbook/Style-guide/README.md (hunk 1 above)
Comment (RuiAAPeres) [reply Cookbook/Style-guide/README.md]: Thanks for your opinion. 👍 :trollface:

Comment (ilyapuchka) [reply Cookbook/Style-guide/README.md]: I double @mluisbrown on that

**Review by ilyapuchka — CHANGES_REQUESTED** [review]
I don't see a strong motivation to forbid (considering our style guide gets translated into SwiftLint rules that will treat all warnings as errors) trailing commas, which are very useful when editing arrays.

**Comment by ilyapuchka** [comment]
I'd like to see this proposal with motivation for it. The fact that we are not putting trailing commas in every array or dictionary we define does not mean that doing so should be a violation of a style guide, and the other way around.
Trailing commas has a huge impact on refactoring experience, for example when rearranging elements or adding new one. It's very easy to forget to add comma and very annoying to remove them in one place and add them in another.

**Comment by TheAdamBorek** [comment]
I don't see any benefits in having this rule. If it is in the code I rather ignore it when I'm reading the code.

Imo this is one of rules where we don't need to listen to majority and if someone is against then ... well ¯\_(ツ)_/¯ maybe it's not worth it.

I'll say again, I don't think it is going to change readablity of our codebase

**Comment by diegopetrucci** [comment]
I will close this for now, and if we have strong feelings about it in the future we can re-open it.

**Comment by andersio** [comment]
Trailing commas do not add any noise that is already there, and enforcing not having it does not appear to have any substantial benefit.

File: Cookbook/Style-guide/README.md (hunk 1 above)
Comment (sergdort) [reply Cookbook/Style-guide/README.md]: I love trailing comas, I forget the comas every time I add new stuff and off course it fails to build

**Pull request closed without merging**